app-ios/tutanota/Sources/Files/FileFacade.swift

```diff
@@ -203,11 +203,30 @@ extension DataTaskResponse {
   init(httpResponse: HTTPURLResponse, encryptedFileUri: String?) {
     self.init(
       statusCode: httpResponse.statusCode,
-      errorId: httpResponse.allHeaderFields["Error-Id"] as! String?,
-      precondition: httpResponse.allHeaderFields["Precondition"] as! String?,
-      suspensionTime:
-        httpResponse.allHeaderFields["Retry-After"] as! String? ?? httpResponse.allHeaderFields["Suspension-Time"] as! String?,
+      errorId: httpResponse.valueForHeaderField("Error-Id"),
+      precondition: httpResponse.valueForHeaderField("Precondition"),
+      suspensionTime: httpResponse.valueForHeaderField("Retry-After") ?? httpResponse.valueForHeaderField("Suspension-Time"),
       encryptedFileUri: encryptedFileUri
     )
   }
 }
+
+fileprivate func getHeader(_ key: String, httpResponse: HTTPURLResponse) -> String? {
+  return (httpResponse.allHeaderFields[key] ?? httpResponse.allHeaderFields[key.lowercased()]) as! String?
+}
+
```

Comment:
I don't see that this function is used, so maybe we can remove it.

Reply:
whoops i meant to do that already, done now